concent_api/core/message_handlers.py

```diff
@@ -193,14 +192,10 @@ def handle_send_ack_report_computed_task(client_message: message.tasks.AckReport
                 task_to_compute,
                 deserialize_message(subtask.task_to_compute.data.tobytes()),
             ])
-            new_report_computed_task = None
-            try:
-                validate_all_messages_identical([
-                    report_computed_task,
-                    deserialize_message(subtask.report_computed_task.data.tobytes()),
-                ])
-            except ConcentValidationError:
-                new_report_computed_task = report_computed_task
+            new_report_computed_task = substitute_new_report_computed_task_if_needed(
+                report_computed_task=report_computed_task,
```

Comment:
report_computed_task -> report_computed_task_from_acknowledgement
data (bytes) -> stored_report_computed_task (ReportComputedTask)

Reply:
done except of changing type of `stored_report_computed_task`. Function `deserialize_message` expects `bytes`, not `ReportComputedTask`

Reply:
deserialize moved to function called. It is passed two `ReportComputedTask` to function `substitute_new_report_computed_task_if_needed`